server/images.go

```diff
@@ -48,16 +58,42 @@ type Model struct {
 	ParentModel    string
 	AdapterPaths   []string
 	ProjectorPaths []string
-	Template       string
 	System         string
 	License        []string
 	Digest         string
 	Options        map[string]interface{}
 	Messages       []Message
+
+	Template *template.Template
 }
 
-func (m *Model) IsEmbedding() bool {
-	return slices.Contains(m.Config.ModelFamilies, "bert") || slices.Contains(m.Config.ModelFamilies, "nomic-bert")
+func (m *Model) Can(caps ...Capability) bool {
+	for _, cap := range caps {
+		switch cap {
+		case CapCompletion:
+			f, err := os.Open(m.ModelPath)
+			if err != nil {
+				slog.Error("couldn't open model file", "error", err)
+				continue
+			}
+			defer f.Close()
+
+			ggml, _, err := llm.DecodeGGML(f)
```

Comment:
this isn't ideal. it should only need to be decoded once but I don't want to change too much stuff in this pr

Reply:
Possible to add a TODO?